game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -507,6 +510,39 @@ private void resetIsAI() {
     isAI = null;
   }
 
+  @VisibleForTesting
+  public void setHasResource(final String value) throws GameParseException {
+    final String[] s = splitOnColon(value);
+    final int n = getInt(s[0]);
+    if (n < 1) {
+      throw new GameParseException("hasResource must be a positive number" + thisErrorMsg());
+    }
+    if (s.length == 1) {
+      hasResource = new String[] {value, "PUs"};
+    } else {
+      for (int i = 1; i < s.length; i++) {
+        // validate that this resource exists in the xml
+        final Resource r = getData().getResourceList().getResource(s[i]);
+        if (r == null) {
+          throw new GameParseException("No resource called: " + s[i] + thisErrorMsg());
```

Comment:
Method `thisErrorMsg` seems odd as name (a generic error message) and usage (added as part of an error message) seem to contradict each other.
Maybe variable `s[i]` should be put in quotes?